### PR TITLE
HOSTEDCP-1308: Add e2e to ensure SA token not mounted on mgmt workloa…

### PR DESCRIPTION
**Pull request by Patryk-Stefanski**
**What this PR does / why we need it**:   Adds e2e to ensure SA token not mounted on mgmt workloa…

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes # [HOSTEDCP-1308](https://issues.redhat.com/browse/HOSTEDCP-1308)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: test/e2e/util/util.go [hunk 1]
```diff
@@ -1717,3 +1717,27 @@ func EnsureNoHCPPodsLandOnDefaultNode(t *testing.T, ctx context.Context, client
...
+
+	for _, pod := range pods.Items {
+		hasPrefix := false
+		for _, prefix := range expectedKasManagementComponents {
+			if strings.HasPrefix(pod.Name, prefix) {
+				hasPrefix = true
+				break
+			}
+		}
+		if !hasPrefix {
```
Comment (enxebre) [inline test/e2e/util/util.go:1737]: why is this not checking Spec.AutomountServiceAccountToken?

Comment (Patryk-Stefanski) [reply test/e2e/util/util.go:1737]: hmm wasnt aware of it, that makes things cleaner. Thanks

Comment (Patryk-Stefanski) [reply test/e2e/util/util.go:1737]: just looking into it not many pods actually use this. I think keeping it the way it is may not be as clean as checking a spec in a pod but its a more accurate way of checking atm.

I can do some refactoring so the expected pods use spec.AutomountServiceAccountToken but think it falls outside of the scope of this Jira. WDYT @enxebre

**Comment by csrwng** [comment]
/override "Red Hat Trusted App Pipeline / hypershift-operator-main-on-pull-request"

**Pull request merged**